### PR TITLE
chore(deps): upgrade hardhat and rm some unnecessary deps

### PR DESCRIPTION
**Pull request by Flouse**
## Changes
- update @nomicfoundation/hardhat-toolbox
  - "ethers": "^5.4.7",
  - "hardhat": "^2.11.0",
- remove some unnecessary dependencies
- [Refactor Revertal tests](https://github.com/godwokenrises/godwoken-tests/pull/291/commits/e00c90267a6fb725d20d1686a5b44121616f1872)

### CODE REVIEW

File: contracts/package.json [hunk 1]
```diff
@@ -13,16 +13,12 @@
...
   ],
-  "author": "",
   "license": "MIT",
   "devDependencies": {
     "@nomicfoundation/hardhat-toolbox": "^2.0.2",
-    "@nomiclabs/hardhat-web3": "^2.0.0",
-    "@openzeppelin/contracts": "^4.9.2",
-    "@openzeppelin/contracts-upgradeable": "^4.8.3",
-    "@openzeppelin/hardhat-upgrades": "^1.18.1",
-    "hardhat": "^2.17.0",
```
Comment (sunchengzhu) [inline contracts/package.json:24]: You may have deleted `"hardhat": "^2.17.0"` by mistake, `@nomicfoundation/hardhat-toolbox` does not contain hardhat, the following is the package.json generated with `npx hardhat`:
``` json
{
   "name": "hardhat-project",
   "devDependencies": {
     "@nomicfoundation/hardhat-toolbox": "^3.0.0",
     "hardhat": "^2.17.1"
   }
}
```

Comment (sunchengzhu) [inline contracts/package.json:24]: "hardhat": "^2.17.0",

File: contracts/package.json [hunk 2]
```diff
@@ -13,16 +13,12 @@
...
-    "web3": "^1.7.1"
+    "@openzeppelin/contracts": "^4.9.3",
```
Comment (sunchengzhu) [inline contracts/package.json:19]:
```suggestion
    "hardhat": "^2.17.1",
    "@openzeppelin/contracts": "^4.9.3",
```

**Review by sunchengzhu — CHANGES_REQUESTED** [review]
please check

File: contracts/package.json [hunk 3]
```diff
@@ -13,16 +13,12 @@
     "Solidity",
     "Hardhat"
   ],
-  "author": "",
   "license": "MIT",
   "devDependencies": {
     "@nomicfoundation/hardhat-toolbox": "^2.0.2",
```
Comment (sunchengzhu) [inline contracts/package.json:18]: Why not upgrade `@nomicfoundation/hardhat-toolbox `to the latest `3.0.0` version?

Comment (Flouse) [reply contracts/package.json:18]: @nomicfoundation/hardhat-toolbox@3 is incompatible with the current test cases.

Could you have a try, and analyze if we can upgrade to hardhat-toolbox@3 in another PR?

File: contracts/package.json (hunk 1 above)
Comment (Flouse) [reply contracts/package.json:24]: updated: https://github.com/godwokenrises/godwoken-tests/pull/291/commits/3ccc6761848d01ceb238d3e7078f73c31be23501

File: contracts/package.json (hunk 3 above)
Comment (sunchengzhu) [reply contracts/package.json:18]: > Could you have a try, and analysis if we can upgrade to hardhat-toolbox@3 in another PR?

Ok, if I have time, I will upgrade to hardhat-toolbox@3 and try to solve the existing problems.

**Review by sunchengzhu — APPROVED** [review]

**Pull request merged**